R/tags.R

```diff
@@ -0,0 +1,125 @@
+
+#' Get a simple data.frame listing all Tags on server
+#' 
+#' @param src The source object
+#' @param use_cache use the tag list previously queried 
+#' from the connect server. src$tags object is null then
+#' this parameter will be ignored
+#' 
+#' @export
+get_tags <- function(src, use_cache = FALSE){
+  validate_R6_class("Connect", src)
+  
+  res <- src$get_tags(use_cache = use_cache)
+  
+  return(res)
+}
+
+#' Get a tibble with more information for each tag
+#' 
+#' @param src The source object
+#' 
+#' @export
+get_tag_info <- function(src){
+  validate_R6_class("Connect", src)
+  
+  res <- src$get_tag_tree()
+  
+  tag_tbl <- parse_tags_tbl(res)
+  
+  return(tag_tbl)
+}
+
+#' Print a visualization of the tag tree
+#' 
+#' @param src The source object
+#' @param top_tag The value to be printed at the top of the 
+#' tag tree. This value can not have any "/" characters in it. 
+#' 
+#' @export
+get_tag_tree <- function(src, top_tag = "RStudio Connect Tags"){
+  validate_R6_class("Connect", src)
+  
+  res <- src$get_tag_tree()
+  
+  tag_tree(res, top_tag = top_tag)
+}
+
+tag_tree <- function(tags, top_tag = "tags"){
+  parsed_tags <- parse_tags(tags, top_tag = top_tag)
+  
+  tag_split <- split(parsed_tags, dirname(parsed_tags))
+  
+  ch <- fs:::box_chars()
```

Comment:
Unfortunately using unexported functions is a no-no, so we'll probably have to see if it's easy enough to pull these little utilities across into our package. It won't be the first utilities used from another package 😅 Although this reminds me that I should probably put attribution somewhere... 😬

Reply:
I removed dependencies with https://github.com/rstudio/connectapi/pull/37/commits/d7e7446eb7dd4ce1eb73ed19e649f29365b49725 

Curiously, is the "no unexported functions" due to the fact that they may change without any notice? I had never heard it before but it makes sense now that you say that.